### PR TITLE
Clarify the representation of `icmp` output

### PR DESCRIPTION
**Pull request by Kmeakin**
Add documentation to `icmp` clarifying the representation of true/false values for scalar and vector comparisons

### CODE REVIEW

File: cranelift/codegen/meta/src/shared/instructions.rs [hunk 1]
```diff
@@ -1615,6 +1615,13 @@ pub(crate) fn define(
 
         When this instruction compares integer vectors, it returns a vector of
         lane-wise comparisons.
+
+        When comparing scalars, the result is `1` if the condition holds, and
```
Comment (kevaundray) [inline cranelift/codegen/meta/src/shared/instructions.rs:1619]: Just a formatting preference:

When comparing scalars, the result is:

   - `1` if the condition holds.
   - `0` if the condition does not hold.
    
When comparing vectors, the result is 
   - `-1` (ie all ones) in each lane if the condition holds.
   -  `0` in each lane if the condition does not hold.

**Comment by jameysharp** [comment]
I think this is a very helpful improvement. I believe the same description would also apply to `fcmp`, if you want to add it there too.

This is an extremely nit-picky detail but I'd prefer "i.e." written with periods. It doesn't actually matter but it bothers me anyway. :sweat_smile:

I'd like @cfallin's confirmation that this description is correct but I believe it is.

**Comment by kevaundray** [comment]
> I think this is a very helpful improvement. I believe the same description would also apply to `fcmp`, if you want to add it there too.
> 
> 
> 
> This is an extremely nit-picky detail but I'd prefer "i.e." written with periods. It doesn't actually matter but it bothers me anyway. :sweat_smile:
> 
> 
> 
> I'd like @cfallin's confirmation that this description is correct but I believe it is.

Now that you've brought it to my attention, I can no longer ignore the missing periods 😅

**Comment by cfallin** [comment]
> I'd like @cfallin's confirmation that this description is correct but I believe it is.

LGTM!

**Review by jameysharp — APPROVED** [review]
Hooray, let's merge it!

**Pull request merged**